pkg/controller/consoleservice/consoleservice_controller.go

```diff
@@ -228,6 +234,13 @@ func (r *ReconcileConsoleService) Reconcile(request reconcile.Request) (reconcil
 	}
 	requeue = requeue || result.Requeue
 
+	// prometheus rules
+	result, err = r.reconcilePrometheusRule(ctx, consoleservice)
```

Comment:
I think we need to detect if the API for the CRD is available or not, similar to how we detect if ConsoleLink or Route is available for instance (somewhere in this same file).

Reply:
:+1: Done